client/components/masonry-wave/index.jsx

```diff
@@ -0,0 +1,91 @@
+import './style.scss';
+
+import { useState, useEffect } from 'react';
+
+function getWindowDimensions() {
+	const { innerWidth: width, innerHeight: height } = window;
+	return {
+		width,
+		height,
+	};
+}
+
+function useWindowDimensions() {
+	const [ windowDimensions, setWindowDimensions ] = useState( getWindowDimensions() );
+
+	useEffect( () => {
+		function handleResize() {
+			setWindowDimensions( getWindowDimensions() );
+		}
+
+		window.addEventListener( 'resize', handleResize );
+		return () => window.removeEventListener( 'resize', handleResize );
+	}, [] );
+
+	return windowDimensions;
+}
+
+// fills the columns with images LTR to imitate the CSS grid behavior
+const fillColumnsWithImages = ( amountOfColumns, images ) => {
+	const columns = Array( amountOfColumns );
+	let columnIndex = 0;
+
+	images.forEach( ( image ) => {
+		if ( columnIndex === amountOfColumns ) {
+			columnIndex = 0;
+		}
+
+		if ( ! columns[ columnIndex ] ) {
+			columns[ columnIndex ] = [];
+		}
+
+		columns[ columnIndex ].push( image );
+
+		columnIndex++;
+	} );
+
+	return columns;
+};
+
+const MasonryWaveElement = ( props ) => {
+	const { image } = props;
+
+	return (
+		<img src={ image?.src } alt={ image?.alt ?? 'alt text' } className="masonry-wave__element" />
+	);
+};
+
+const MasonryWaveColumn = ( props ) => {
+	const { images } = props;
+
+	return (
+		<div className="masonry-wave__column">
+			{ images.map( ( image ) => (
+				<MasonryWaveElement image={ image } />
```

Comment:
@Rolilink I'm seeing some console warnings about missing `key` props on `MasonryWave` and `MasonryWaveColumn`:
`Each child in a list should have a unique "key" prop.`

Reply:
✅  Added the keys for this component